openedx/core/djangoapps/models/course_details.py

```diff
@@ -271,6 +275,7 @@ def update_from_json(cls, course_key, jsondict, user):  # pylint: disable=too-ma
                 and jsondict['self_paced'] != descriptor.self_paced):
             descriptor.self_paced = jsondict['self_paced']
             dirty = True
+            is_pacing_changed = True
```

Comment:
I don't think we need to maintain a `is_pacing_changed` flag.  Inside this if block, we can just send the `COURSE_PACING_CHANGE` signal.